src/Public/js/nipsweb.managelibrary.js

```diff
@@ -1,258 +1,304 @@
+                        file.name + ': <input type="text" class="title" name="' +
+                        response.fileid + '" id="resuploadname-' + i +
+                        '" placeholder="Enter a helpful name..." /></div>');
+
+                    if (window.auxid.match(/^aux-\d+$/)) {
+                        //This is a central one - it can have an expiry
+                        result.append(
+                            $('<input type="text" placeholder="Expiry date" />').addClass('date').attr('id', 'resuploaddate-' + i).datetimepicker({
+                                pickTime: 'false'
+                            }))
+                            .append('<em>Leave blank to never expire</em>&nbsp;&nbsp;');
+                    }
+                    result.append('<div id="confirminator-' + (response.fileid.replace(/\.mp3/, '')) + '"></div>');
+                    result.append($('<a href="javascript:">Save</a>').click(function () {
+                        var title = result.find('input.title').val();
+                        var expire = result.find('input.date').val() || null;
+                        var fileid = result.find('input.title').attr('name');
+
+                        result.html('Adding <em>' + title + '</em> to library');
+                        $.ajax({
+                            url: myury.makeURL('NIPSWeb', 'confirm_aux_upload'),
+                            data: {
+                                auxid: window.auxid,
+                                fileid: fileid,
+                                title: title,
+                                expires: expire
+                            },
+                            dataType: 'json',
+                            type: 'get',
+                            success: function (data) {
+                                data.fileid = data.fileid.replace(/\.mp3/, '');
+                                if (data.status == 'OK') {
+                                    result.removeClass('alert-info')
+                                            .addClass('alert-success')
+                                            .html('<div class="glyphicon glyphicon-ok"></div><em>' + title + '</em> added to library');
+                                } else {
+                                    result.removeClass('alert-info')
+                                            .addClass('alert-danger')
+                                            .html('<div class="glyphicon glyphicon-exclamation-sign"></div><em>' + title + '</em> could not be added to library<br>'
+                                                + data.error);
+                                }
+                            }
+                        });
+                    }));
+
+                    $('#res-result').append(result);
+                }
+            }
+        });
+    };
+
+    var initialise = function() {
+        $('#res-type-sel').on('change', res_type_sel_change_handler);
```

Comment:
change in chrome doesn't fire until the element has been blur'd sometimes - they're needed.